pkg/auth/credentials.go

```diff
@@ -109,11 +125,35 @@ func GetNewCredentials() (cred Credentials, err error) {
 	if err != nil {
 		return cred, err
 	}
-	cred.SecretKey = string([]byte(base64.StdEncoding.EncodeToString(keyBytes))[:secretKeyMaxLen])
+	cred.SecretKey = string([]byte(base64.URLEncoding.EncodeToString(keyBytes))[:secretKeyMaxLen])
+
+	if _, ok := m["exp"]; !ok {
+		cred.Expiration = timeSentinel
+		return cred, nil
+	}
+	expiry, ok := m["exp"].(float64)
+	if !ok {
+		cred.Expiration = timeSentinel
+		return cred, nil
+	}
+
+	m["sub"] = cred.AccessKey
+	jwt := jwtgo.NewWithClaims(jwtgo.SigningMethodHS512, jwtgo.MapClaims(m))
+
+	cred.Expiration = time.Unix(int64(expiry), 0)
+	cred.SessionToken, err = jwt.SignedString([]byte(cred.SecretKey))
```

Comment:
Note that we have to sign sts with admin secret, otherwise it defeats the purpose as the client knows its own secret key